rs/system_api/src/lib.rs

```diff
@@ -3445,14 +3451,36 @@ impl SystemApi for SystemApiImpl {
                 }),
                 Some(request) => {
                     if request.is_timeout_set() {
-                        Err(HypervisorError::ToolchainContractViolation {
+                        return Err(HypervisorError::ToolchainContractViolation {
```

Comment:
If we return here we don't get to the `trace_syscall` below. So I think it needs to remain an `if / else`.

Reply:
It does, thanks for the catch.